luigi/contrib/azureblob.py

```diff
@@ -0,0 +1,286 @@
+            container=container, blob=blob, location=location))
+        return self.connection.get_blob_to_path(container, blob, location)
+
+    def create_container(self, container_name):
+        return self.connection.create_container(container_name)
+
+    def delete_container(self, container_name):
+        lease_id = self.connection.acquire_container_lease(container_name)
+        self.connection.delete_container(container_name, lease_id=lease_id)
+
+    def exists(self, path):
+        container, blob = self.splitfilepath(path)
+        return self.connection.exists(container, blob)
+
+    def remove(self, path, recursive=True, skip_trash=True):
+        container, blob = self.splitfilepath(path)
+        if not self.exists(path):
+            return False
+        lease_id = self.connection.acquire_blob_lease(container, blob)
+        self.connection.delete_blob(container, blob, lease_id=lease_id)
+        return True
+
+    def mkdir(self, path, parents=True, raise_if_exists=False):
+        container, blob = self.splitfilepath(path)
+        if raise_if_exists and self.exists(path):
+            raise FileAlreadyExists("The Azure blob path '{blob}' already exists under container '{container}'".format(
+                blob=blob, contaniner=container))
+
+    def isdir(self, path):
+        """
+        Azure Blob Storage has no concept of directories. It always returns False
+        :param str path: Path of the Azure blob storage
+        :return: False
+        """
+        return False
+
+    def move(self, path, dest):
+        return self.copy(path, dest) and self.remove(path)
+
+    def copy(self, path, dest):
+        source_container, source_blob = self.splitfilepath(path)
+        dest_container, dest_blob = self.splitfilepath(dest)
+        if source_container != dest_container:
+            raise Exception(
+                "Can't copy blob from '{source_container}' to '{dest_container}'. File can be moved within container".format(
+                    source_container=source_container, dest_container=dest_container
+                ))
+
+        source_lease_id = self.connection.acquire_blob_lease(source_container, source_blob)
+        destination_lease_id = self.connection.acquire_blob_lease(dest_container, dest_blob) if self.exists(dest) else None
+        try:
+            return self.connection.copy_blob(source_container, dest_blob, self.connection.make_blob_url(
+                source_container, source_blob),
+                destination_lease_id=destination_lease_id, source_lease_id=source_lease_id)
+        finally:
+            self.connection.release_blob_lease(source_container, source_blob, source_lease_id)
+            if destination_lease_id is not None:
+                self.connection.release_blob_lease(dest_container, dest_blob, destination_lease_id)
+
+    def rename_dont_move(self, path, dest):
+        self.move(path, dest)
```

Comment:
Would it make more sense for this to be `move()` with some default option overwritten?

Reply:
this method `def rename_dont_move(self, path, dest):` overrides the method in `FileSystem` which is implemented in base class as

```python
warnings.warn("File system {} client doesn't support atomic mv.".format(self.__class__.__name__))
        if self.exists(dest):
            raise FileAlreadyExists()
        self.move(path, dest)
```

This implementation is not relevant to azure blob, so I had to override it.

Reply:
Ohhhh. I didn't even look to see it was in Filesystem. Sorry about this unnecessary back and forth

:+1: